### PR TITLE
Implement workflow cancellations

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Implemented workflow cancel machine & sending activation to lang

## Why?
Sometimes users cancel their workflows, ya know.

## Checklist

1. Closes 

2. How was this tested:
New tests

3. Any docs updates needed?

### CODE REVIEW

File: src/machines/workflow_machines.rs [hunk 1]
```diff
@@ -591,6 +605,10 @@ impl WorkflowMachines {
                     let canm = self.add_new_machine(continue_as_new(attrs));
                     self.current_wf_task_commands.push_back(canm);
                 }
+                WFCommand::AckCancelled(attrs) => {
```
Comment (bergundy) [inline src/machines/workflow_machines.rs:608]: Didn't we rename this?

File: src/protos/mod.rs [hunk 2]
```diff
@@ -203,6 +226,9 @@ pub mod coresdk {
                         workflow_command::Variant::ContinueAsNewWorkflowExecution(_) => {
                             write!(f, "ContinueAsNewWorkflowExecution")
                         }
+                        workflow_command::Variant::AckExecutionCancelled(_) => {
```
Comment (bergundy) [inline src/protos/mod.rs:229]: Same here, let's rename

File: src/protos/mod.rs [hunk 3]
```diff
@@ -203,6 +226,9 @@ pub mod coresdk {
...
+                            write!(f, "AckExecutionCancelled")
```
Comment (bergundy) [inline src/protos/mod.rs:230]: And this string

**Review by bergundy — COMMENTED** [review]
In the Node SDK I'm going to have these test scenarios:
1. Workflow cancel requested -> workflow is immediately cancelled
2. Workflow cancel requested -> workflow immediately completes
3. Workflow cancel requested -> workflow immediately fails
4. Workflow cancel requested -> workflow completes after cleanup
5. Workflow cancel requested -> workflow is cancelled after cleanup
6. Workflow cancel requested -> workflow fails after cleanup

These are all of the possible scenarios I can think of, would you test these here? Is this overkill?

File: src/machines/workflow_machines.rs (hunk 1 above)
Comment (Sushisource) [reply src/machines/workflow_machines.rs:608]: This is a core-internal name but yeah I'll keep it consistent

File: protos/local/workflow_activation.proto [hunk 4]
```diff
@@ -99,7 +99,7 @@ message QueryWorkflow {
 
 /// Cancel a running workflow
 message CancelWorkflow {
-    // TODO: add attributes here
+    repeated common.Payload details = 1;
```
[the remark this answers is not in the document]
Comment (Sushisource) [reply protos/local/workflow_activation.proto:102]: The cancellation request. Will add docstring

**Comment by Sushisource** [comment]
@bergundy Re: testing. I definitely would *like* to, but as I was mentioning to you in slack it's amazing how much time trying to get the baby Rust SDK/worker going takes compared to everything else. So from a practicality standpoint, core getting tested via your tests just makes a lot of sense, and maybe we can defer the closer-to-core testing until we start productionalizing a Rust SDK.

If your tests uncover problems in Core, then adding regression tests for those scenarios has worked well so far.

I am definitely open to investing more time up front, too, just a question of what we feel is the best investment of our time ATM

**Review by bergundy — APPROVED** [review]
Awesome, let's merge this

**Pull request merged**